internal/api/schema.go

```diff
@@ -100,14 +116,20 @@ func (a *API) GetSchema(ctx context.Context, in *stencilv1beta1.GetSchemaRequest
 }
 
 func (a *API) HTTPGetSchema(w http.ResponseWriter, req *http.Request, pathParams map[string]string) (*schema.Metadata, []byte, error) {
+	txn := newrelic.FromContext(req.Context())
+	newTxn := newRelic.NewTransaction(txn)
+	segment := newTxn.StartGenericSegment("GetSchema")
 	namespaceID := pathParams["namespace"]
 	schemaName := pathParams["name"]
 	versionString := pathParams["version"]
 	v, err := strconv.ParseInt(versionString, 10, 32)
 	if err != nil {
+		segment.End()
```

Comment:
actually we don't need to call for each checks. 
we can just use  defer
```
	endFunc := newTxn.StartGenericSegment("Compatibility checker")
	defer endFunc()
```

Reply:
I have made the changes. If it looks good please mark the thread "resolved".